lib/database.js

```diff
@@ -57,12 +58,12 @@ function update_address(hash, txid, amount, type, cb) {
           var tx_array = address.txs;
           var received = address.received;
           var sent = address.sent;
-          if (type == 'vin') {
-            sent = sent + amount;
-          } else {
-            received = received + amount;
-          }
-          if (unique == true) {
+          if ((unique == true) || (type != tx_array[index].type)) {
```

Comment:
There was some duplicated code here, have stripped it down as it wasn't catching tx's sent to self properly